waves-integration-it/src/test/scala/com/wavesplatform/dex/grpc/integration/ExchangeTransactionMinimumFeeTestSuite.scala

```diff
@@ -0,0 +1,176 @@
+
+  val scriptedAmount: IssuedAsset = IssuedAsset(issueScriptedAmountTx.id())
+  val scriptedPrice: IssuedAsset = IssuedAsset(issueScriptedPriceTx.id())
+
+  val scriptedAmountWavesPair: AssetPair = AssetPair(scriptedAmount, Waves)
+  val scriptedAmountScriptedPricePair: AssetPair = AssetPair(scriptedAmount, scriptedPrice)
+  val btcScriptedPricePair: AssetPair = AssetPair(btc, scriptedPrice)
+
+  "Node should set correct fee for Exchange transaction when" - {
+
+    "amount asset is scripted" in broadcastAndValidate(
+      mkExchange(alice, alice, scriptedAmountWavesPair, 1.btc, 1.waves, matcher = alice, matcherFee = 0.007.waves),
+      0.007.waves
+    )
+
+    "price asset is scripted" in broadcastAndValidate(
+      mkExchange(alice, alice, btcScriptedPricePair, 10.waves, 1.btc, matcher = alice, matcherFee = 0.007.waves),
+      0.007.waves
+    )
+
+    "fee asset of sell order is scripted" in broadcastAndValidate(
+      mkExchange(alice, alice, wavesUsdPair, 10.waves, 1.usd, matcher = alice, sellOrderFeeAsset = scriptedAmount),
+      0.003.waves
+    )
+
+    "fee asset of buy order is scripted" in broadcastAndValidate(
+      mkExchange(alice, alice, wavesUsdPair, 10.waves, 1.usd, matcher = alice, buyOrderFeeAsset = scriptedPrice),
+      0.003.waves
+    )
+
+    "sell order owners' account is scripted" in broadcastAndValidate(
+      mkExchange(alice, scriptedAccount, wavesUsdPair, 10.waves, 1.usd, matcher = alice),
+      0.003.waves
+    )
+
+    "buy order owners' account is scripted" in broadcastAndValidate(
+      mkExchange(scriptedAccount, alice, wavesUsdPair, 10.waves, 1.usd, matcher = alice),
+      0.003.waves
+    )
+
+    "buy and sell orders owners, amount and price assets, matcher and fee assets of both orders are scripted" in broadcastAndValidate(
+      mkExchange(
+        scriptedAccount,
+        scriptedAccount,
+        scriptedAmountScriptedPricePair,
+        10.waves,
+        1.usd,
+        matcher = scriptedAccount,
+        sellOrderFeeAsset = scriptedAmount,
+        buyOrderFeeAsset = scriptedPrice,
+        matcherFee = 0.015.waves
+      ),
+      0.015.waves
+    )
+  }
+
+  def broadcastAndValidate(tx: Transaction, expectedFee: Long): Unit = {
+    broadcastAndAwait(tx)
+    wavesNode1.api.transactionInfo(tx.id()).fee().value() shouldBe expectedFee
+  }
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    broadcastAndAwait(IssueUsdTx, IssueBtcTx, issueScriptedAmountTx, issueScriptedPriceTx)
+
+    broadcastAndAwait(
+      mkTransfer(alice, scriptedAccount.toAddress, 100.usd, usd),
+      mkTransfer(bob, alice.toAddress, 100.btc, btc),
+      mkTransfer(bob, scriptedAccount.toAddress, 100.btc, btc),
+      mkTransfer(alice, scriptedAccount.toAddress, 1000.waves, Waves),
+      mkTransfer(alice, scriptedAccount.toAddress, 1000.waves, scriptedAmount, 0.007.waves),
+      mkTransfer(alice, scriptedAccount.toAddress, 1000.waves, scriptedPrice, 0.007.waves)
```

Comment:
We don't need ".toAddress", because we have an implicit conversion

Reply:
1) `.toAddress`-- fixed
2) Matcher account is scripted and fit 200 complexity, then Node doesn't require an additional fee  -- if matcher has script with less than 200 complexity the node requires extra fee
3) Matcher account is scripted and don't fit 200 complexity, then Node requires an additional fee  -- added